### PR TITLE
Refactor cross-cutting stores from "options/" into "store/" directory

### PR DESCRIPTION
**Pull request by twschiller**
Part of the work of cleaning up warnings from: https://github.com/pixiebrix/pixiebrix-extension/pull/2276. The idea is that contexts (options page, contentScript, etc.) shouldn't import from each other directly (they should communicate via the messenger API). If there are shared code, the shared code should be extracted

Changes
- Moves all the slices/stores/persist logic from the "options/" folder to the "/store" folder

Naming Convention:
- Renames "optionsSlice" to "extensionsSlice" as the export. There's still some places within files where it's referred to as options (property names, etc.)

Design Questions:
- Should the workshopSlice be in the options folder since it's only used by options page?

### CODE REVIEW

File: src/options/store.ts [hunk 1]
```diff
@@ -67,7 +65,10 @@ if (process.env.NODE_ENV === "development") {
 const store = configureStore({
   reducer: {
     router: connectRouter(hashHistory),
-    options: persistReducer(persistOptionsConfig, optionsSlice.reducer),
+    options: persistReducer(
+      persistExtensionOptionsConfig,
+      extensionsSlice.reducer
+    ),
     services: persistReducer(persistServicesConfig, servicesSlice.reducer),
     // XXX: settings and workshop use the same persistor config?
```
Comment (twschiller) [inline src/options/store.ts:73]: This is still wacky - I didn't look into this as part of this PR

**Comment by BLoe** [comment]
>Should the workshopSlice be in the options folder since it's only used by options page?

If it's going to remain a separate slice, I would say keep it where it is alongside the other slices.

**Review by BLoe — APPROVED** [review]
I'm not as familiar with the options page and workshop redux stuff, but these changes seem fine to me.

**Review by BALEHOK — APPROVED** [review]

**Comment by twschiller** [comment]
Capturing another alternative here for future consideration/experimentation:

We could create a "shared" folder for cross-cutting concerns. The potential downside is search/file navigation issues when too many files have the same name (e.g., selectors.ts)

```
shared/
  store/
    extensions/
      slice.ts
      selectors.ts
    services/
      slice.ts
      selectors.ts
```

**Review by fregante — COMMENTED** [review]
I'm a bit late but from a quick look it LGTM. 💯

**Comment by fregante** [comment]
I'm still seeing `/options/store` imports in the action panel, it's the last one:

options/store

**Pull request merged**